tests/52user-directory/01public.pl

```diff
@@ -44,17 +55,22 @@
       my ( $user ) = @_;
 
       my $room_id;
+      my $searching_user;
 
       my $displayname = generate_random_displayname();
+      my $searching_displayname = generate_random_displayname();
+
+      matrix_set_displayname( $user, $displayname )->then( sub {
+         matrix_create_user_on_server( $user->http,
+            displayname => $searching_displayname
+         );
+      })->then( sub {
+         ( $searching_user ) = @_;
 
-      matrix_set_displayname( $user, $displayname )
-      ->then( sub {
          matrix_create_room( $user,
             preset => "private_chat",
          );
       })->then( sub {
-         ( $room_id ) = @_;
-
          matrix_get_user_dir_synced( $user, $displayname );
```

Comment:
this user still seems to be looking for themselves?

Reply:
fixed